src/main/java/com/taxi/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,301 @@
+package com.taxi.dao;
+
+import com.taxi.dao.interfaces.CarDao;
+import com.taxi.exception.DataProcessingException;
+import com.taxi.model.Car;
+import com.taxi.model.Driver;
+import com.taxi.model.Manufacturer;
+import com.taxi.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class CarDaoImpl implements CarDao {
+    private final ConnectionUtil connectionUtil;
+
+    @Autowired
+    public CarDaoImpl(ConnectionUtil connectionUtil) {
+        this.connectionUtil = connectionUtil;
+    }
+
+    @Override
+    public Car create(Car car) {
+        Connection connection = connectionUtil.getConnection();
+        try {
+            connection.setAutoCommit(false);
+            createCarInDb(car, connection);
+            connection.commit();
+        } catch (SQLException e) {
+            try {
+                connection.rollback();
+            } catch (SQLException ex) {
+                throw new RuntimeException("Rollback is not successful", ex); // TODO: прибрати це, і додати логгер?
+            }
+            throw new DataProcessingException("Can't create car: "
+                    + car, e);
+        } finally {
+            try {
+                connection.setAutoCommit(true);
+                connection.close();
+            } catch (SQLException e) {
+                throw new RuntimeException("Couldn't close the connection", e);
+            }
+        }
+        return car;
+    }
+
+    private void createCarInDb(Car car, Connection connection) throws SQLException {
```

Comment:
Стараемся не делать void методы. Тем более для создания традиционно принято возвращать созданный объект
```suggestion
    private Car createCarInDb(Car car, Connection connection) throws SQLException {
```